os/fs/smartfs/smartfs_utils.c

```diff
@@ -1211,22 +1230,21 @@ int smartfs_find_availableentry(struct smartfs_mountpt_s *fs, uint16_t *parentdi
 				direntry->dsector = psector;
 				direntry->doffset = offset;
 				direntry->flags = entry->flags;
-				*parentdirsector = direntry->dsector;
+				direntry->last_dsector = direntry->dsector;
```

Comment:
please move this to between line 1230~1231 and below
```
/* initialize dsector & last_dsector as psector, dsector will be changed to new entry sector if current entry sector is full */
direntry->dsector = psector;
direntry->last_dsector = psector;

Reply:
If this part of code is reached
It means entry has been found. 
So this comment is not right.
After this line, function will return and not look further.

Reply:
Dsector will not need to be changed because: -
Line 1229 `if (IS_AVAILABLE_ENTRY(entry)) {`